curp/src/client.rs

```diff
@@ -116,9 +282,15 @@ where
     ) -> Result<(Option<<C as Command>::ER>, bool), CommandProposeError<C>> {
         debug!("fast round for cmd({}) started", cmd_arc.id());
         let req = ProposeRequest::new(cmd_arc.as_ref()).map_err(Into::<ProposeError>::into)?;
-        let mut rpcs: FuturesUnordered<_> = self
-            .connects
-            .values()
+
+        let (connects, superquorum) = self.state.map_read(|state_r| {
```

Comment:
Is this a good practice here? I'm afraid that doing so may be a disservice to the performance in that it might exacerbate our lock contention that are already severe.

Reply:
This does disservice some performance, but the old implementation can’t meet the needs of membership changes, do you have a better suggestion?

Reply:
We can move the `connects` back to the client, but implement it with some lock-free hashmap, like dashmap.

Reply:
Fixed